prepare/kotlin-compiler-internal-test-framework/build.gradle.kts

```diff
@@ -1,21 +1,55 @@
+import org.gradle.jvm.tasks.Jar
+
 plugins {
     java
+    `maven-publish`
 }
 
 val embedded by configurations
 
 dependencies {
-    embedded(projectTests(":compiler:tests-common")) { isTransitive = false }
+    embedded(project(":compiler:serialization")){ isTransitive = false }
+    embedded(projectTests(":compiler:tests-common-jvm6")){ isTransitive = false }
+    embedded(project(":kotlin-scripting-compiler-impl")){ isTransitive = false }
+    embedded(projectTests(":compiler:test-infrastructure")){ isTransitive = false }
+    embedded(projectTests(":compiler:test-infrastructure-utils")){ isTransitive = false }
+    embedded(projectTests(":compiler:tests-compiler-utils")){ isTransitive = false }
+    embedded(projectTests(":compiler:tests-common-new")) { isTransitive = false }
 }
 
 publish()
+noDefaultJar()
 
-runtimeJar()
+val runtimeJar: TaskProvider<out Jar> = runtimeJar(unshadedCompiler())
+val runtimeConfig by configurations.creating
+artifacts {
```

Comment:
This setup should not be required after you called publish

Reply:
I'm not sure if I understood your suggestion correctly. I think the idea was to include the compiler modules directly using the compilerModules array and add those to the source jar and then publish the source jar (no customization necessary in this case)? When I did that the source jar was not published at all.